sklearn/impute.py

```diff
@@ -452,17 +452,19 @@ class ChainedImputer(BaseEstimator, TransformerMixin):
         "random"
             A random order for each round.
 
-    n_imputations : int, optional (default=100)
-        Number of chained imputation rounds to perform, the results of which
-        will be used in the final average.
-
-    n_burn_in : int, optional (default=10)
-        Number of initial imputation rounds to perform the results of which
-        will not be returned.
+    n_iter : int, optional (default=10)
+        Number of imputation rounds to perform before returning the final
+        imputations.
 
     predictor : estimator object, default=BayesianRidge()
         The predictor to use at each step of the round-robin imputation.
-        It must support ``return_std`` in its ``predict`` method.
+        It must support ``return_std`` in its ``predict`` method if
+        ``sample_after_predict`` option is set to ``True`` below.
+
+    sample_after_predict : boolean, default=False
```

Comment:
I think this name is unnecessarily verbose.

Brainstorming.

* fill_with='expectation' vs 'sample'
* random_impute bool
* use_std
* sampled
* random_draw
* randomized (but there are other random components)

Reply:
How about `predict_posterior` bool?